packages/messageport/tizen/src/messageport_manager.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "messageport_manager.h"
+
+#include <bundle.h>
+#include <flutter/standard_message_codec.h>
+
+#include <cstdint>
+#include <vector>
+
+#include "log.h"
+
+MessagePortManager::MessagePortManager() {}
+
+MessagePortManager::~MessagePortManager() {
+  for (auto& pair : local_ports_) {
+    int port = pair.second;
+    bool is_trusted = pair.first.second;
+    int ret = MESSAGE_PORT_ERROR_NONE;
+    if (is_trusted) {
+      ret = message_port_unregister_trusted_local_port(port);
+    } else {
+      ret = message_port_unregister_local_port(port);
+    }
+
+    if (MESSAGE_PORT_ERROR_NONE != ret) {
+      LOG_ERROR("Failed to unregister port: %s", get_error_message(ret));
+    }
+  }
+}
+
+static bool ConvertEncodableValueToBundle(flutter::EncodableValue& value,
+                                          bundle* bundle) {
+  if (!bundle) {
+    return false;
+  }
+  std::unique_ptr<std::vector<uint8_t>> encoded =
+      flutter::StandardMessageCodec::GetInstance().EncodeMessage(value);
```

Comment:
If a non-Flutter app is to communicate with a Flutter app via the message port, should it also use the same codec (`StandardMessageCodec`)?

Reply:
I think so.since this is a kind of serialization, it necessarily needs a deserialization algorithm.